ql/src/java/org/apache/hadoop/hive/ql/exec/ColumnStatsDropTask.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.exec;
+
+import org.apache.hadoop.hive.ql.ErrorMsg;
+import org.apache.hadoop.hive.ql.plan.ColumnStatsDropWork;
+import org.apache.hadoop.hive.ql.plan.api.StageType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * ColumnStatsDropTask implementation. Examples:
+ * ALTER TABLE src_stat DROP STATISTICS for columns [comma separated list of columns];
+ * ALTER TABLE src_stat_part PARTITION(partitionId=100) DROP STATISTICS for columns [comma separated list of columns];
+ **/
+
+public class ColumnStatsDropTask extends Task<ColumnStatsDropWork> {
+  private static final long serialVersionUID = 1L;
+  private static final Logger LOG = LoggerFactory.getLogger(ColumnStatsDropTask.class);
+
+  @Override
+  public int execute() {
+    try {
+      getHive()
+          .deleteColumnStatistics(work.getDbName(), work.getTableName(), work.getPartName(), work.getColNames());
+      return 0;
+    } catch (Exception e) {
+      setException(e);
+      LOG.info("Failed to drop column stats in metastore", e);
+      return ErrorMsg.getErrorMsg(e.getMessage()).getErrorCode();
```

Comment:
What is the error code in this case?
What type of exceptions do we expect?
Should we create a unique error?

Reply:
I think this is catching any `Exception` that may come from HMS.
I can create a new error code for this and add it to the `HiveException` in `Hive#deleteColumnStatistics`

Reply:
Does the exceptions coming from HMS are general errors (like i/o error) or there are exceptions related to the drop stats functionality. If the latter is true I think we should introduce an error code for those ones and `GENERIC_ERROR` is fine otherwise .

Reply:
I think we can just use `GENERIC_ERROR` then. I will make the changes.